File: src/DurableDice.Common/Models/State/Player.cs
using DurableDice.Common.Enums;
using Newtonsoft.Json;

namespace DurableDice.Common.Models.State;

[JsonObject(MemberSerialization.OptIn)]
public class Player
{
    [JsonProperty]
    public int Index { get; set; }

    [JsonProperty]
    public string Id { get; set; } = "";

    [JsonProperty]
    public string Name { get; set; } = "";

    [JsonProperty]
    public int DiceBuffer { get; set; }

    [JsonProperty]
    public int DiceMovesThisTurn { get; set; }

    [JsonProperty]
    public int ContinuousFieldCount { get; set; }

    [JsonProperty]
    public bool IsReady { get; set; }

    [JsonProperty]
    public BotType? BotType { get; set; }
}

File: src/DurableDice.GameClientV2/Player.fs
module Player

open Theme

type Player =
    { Index: int
      Id: string
      Name: string
      DiceBuffer: int
      DiceMovesThisTurn: int
      ContinuousFieldCount: int
      IsReady: bool
      BotType: int option }

module Player = 
    let isPlayer (playerId) (player) = player.Id = playerId
    let color (player) =
        Theme.color (Some player.Index)
    
